src/backend/binaryen/utils.ts

```diff
@@ -1814,6 +1818,21 @@ export namespace FunctionalFuncs {
                 ) {
                     res = module.i32.eqz(res);
                 }
+                if (
+                    opKind === ts.SyntaxKind.LessThanEqualsToken ||
+                    opKind === ts.SyntaxKind.LessThanToken ||
+                    opKind === ts.SyntaxKind.GreaterThanEqualsToken ||
+                    opKind === ts.SyntaxKind.GreaterThanToken
+                ) {
+                    const ret = module.call(
+                        dyntype.dyntype_to_number,
```

Comment:
`any` can not only compare with `number`, but also can compare with `string` etc.

like: 
```ts
const a: any = 'hello';
const b = 'hi';
if (a > b) {
    console.log('jj')
} else {
    console.log('kk')
}
```

So we should also handle differently by the operator type kind.

Reply:
done